shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -72,13 +75,57 @@
     private UUID writerIdPerformingAppends;
     private int currentBlockSize;
     private int bytesLeftInBlock;
+    private final List<Session> pendingWrites = new LinkedList<>();
```

Comment:
I think `ArrayList` would be more efficient than a `LinkedList`

Reply:
yes .. done